### PR TITLE
Tw 269 request attributes

### PR DESCRIPTION
**Pull request by BirgitBader**
for carts and categories

@ubep please check and merge

### CODE REVIEW

File: apps/data-types.md [hunk 1]
```diff
@@ -52,26 +52,81 @@ This object is used for the attributes of shippingAddress and billingAddress.
...
+| firstName | string | The first name of the person associated with the address.|
+| lastName | string | The last name of the person associated with the address.|
+| street | string | The street name of the address. |
+| zipCode | string | The zip or postal code of the address. |
+| city | string | The name of the city. |
+| state | string | The name of the state.|
+| country | string | The name of the country. |
+| birthday | string | The date of birth of the person associated with the address. |
+| emailAddress | string | The email address of the person associated with the address. |
+
```
Comment (ubep) [inline apps/data-types.md]: missing address attributes:
- company
- salutation
- streetDetails
- title
- vatId

File: apps/data-types.md [hunk 2]
```diff
@@ -52,26 +52,81 @@ This object is used for the attributes of shippingAddress and billingAddress.
...
+## cartPostRequest
+
+| Attribute | Type | Description |
+| - | :-: |  - |
+| currency | string | The currency code of the price according to ISO 4217.|
+| taxType | string | Indicates if the amount includes tax, e.g. gross.|
```
Comment (ubep) [inline apps/data-types.md]: taxType, can be GROSS, NET or NONE

File: apps/data-types.md [hunk 3]
```diff
@@ -52,26 +52,81 @@ This object is used for the attributes of shippingAddress and billingAddress.
...
+| locale | string | The locale that identifies the origin of the customer.|
+| lineItems | array of [cartIdLineItemPostRequest](page:apps-data-types#cartidlineitempostrequest)  | The product line items in the basket.|
+
+## cartIdLineItemPostRequest
+
+| Attribute | Type | Description |
+| - | :-: |  - |
+| productId | string | The unique identifier of the product.|
+| quantity | string | The quantity of the product line item.|
```
Comment (ubep) [inline apps/data-types.md]: quantity is decimal

File: apps/data-types.md [hunk 4]
```diff
@@ -52,26 +52,81 @@ This object is used for the attributes of shippingAddress and billingAddress.
...
+
+## cartLineItemIdPutRequest
```
Comment (ubep) [inline apps/data-types.md]: cartLineItemPutRequest - without 'Id'

File: apps/data-types.md [hunk 5]
```diff
@@ -52,26 +52,81 @@ This object is used for the attributes of shippingAddress and billingAddress.
 | minCartValue | object of [price](page:apps-data-types#price) | The minimum order value of a shop. |
 | checkoutURL | string | The URL that redirects the browser to the merchant’s shop in order to complete the checkout. |
 
+## cartIdAddressPutRequest
```
Comment (ubep) [inline apps/data-types.md]: delete this one, we can use *address* instead

File: apps/data-types.md [hunk 6]
```diff
@@ -52,26 +52,81 @@ This object is used for the attributes of shippingAddress and billingAddress.
...
+| lastName | string | The last name of the person associated with the address.|
+| street | string | The street name of the address. |
+| zipCode | string | The zip or postal code of the address. |
+| city | string | The name of the city. |
+| state | string | The name of the state.|
+| country | string | The name of the country. |
+| birthday | string | The date of birth of the person associated with the address. |
+| emailAddress | string | The email address of the person associated with the address. |
+
+## cartPostRequest
```
Comment (ubep) [inline apps/data-types.md]: rename to cartRequest

File: apps/data-types.md [hunk 7]
```diff
@@ -52,26 +52,81 @@ This object is used for the attributes of shippingAddress and billingAddress.
...
+
+| Attribute | Type | Description |
+| - | :-: |  - |
+| currency | string | The currency code of the price according to ISO 4217.|
+| taxType | string | Indicates if the amount includes tax, e.g. gross.|
+| locale | string | The locale that identifies the origin of the customer.|
+| lineItems | array of [cartIdLineItemPostRequest](page:apps-data-types#cartidlineitempostrequest)  | The product line items in the basket.|
+
+## cartIdLineItemPostRequest
```
Comment (ubep) [inline apps/data-types.md]: rename to *productLineItemRequest* or *productLineItem (request)*

File: apps/data-types.md [hunk 8]
```diff
@@ -52,26 +52,81 @@ This object is used for the attributes of shippingAddress and billingAddress.
...
-| pageTitle | string | The title of this category. |
+| pageTitle | string | The page title of this category. |
 | description | string | The description of the category. |
 | specialOffer | boolean | Special offers of this category. |
 | images | array of [image](page:apps-data-types#image) | The images belonging to this category. |
 | parent | [link](page:apps-data-types#link) | The link to the parent category. |
 | subCategories | array of [link](page:apps-data-types#link) | A list of links to the subcategories. |
 | sfUrl | string | The link to the categories in the shop’s storefront. |
 
+## categoryIdPutRequest
```
Comment (ubep) [inline apps/data-types.md]: rename to *category (update request)*

File: apps/data-types.md [hunk 9]
```diff
@@ -83,15 +138,19 @@ This object is used for the attributes of shippingAddress and billingAddress.
...
-| key | string | The identifier of the custom attribute. |
-| displayKey | string | The displayed name of the custom attribute. |
-| singleValue | boolean | Indicates if just one feature is selected for the custom attribute.  |
-| type | enum | The data type of the custom attribute. Can be *string*, *number*, *bool*, *datetime*, *time*, *url*.|
-| values | array of [variationValue](page:apps-data-types#variationvalue)| The options selected for the custom attribute. |
+| op | string | Patch operation to perform. Allows `add` and `remove`. *customerNumber* supports `add` only.|
+| path | string | Contains the value that references the location where the operation is performed. |
+| value | object | New value to apply. Only required for `add` operations. |
+
+## customerPostRequest
```
Comment (ubep) [inline apps/data-types.md]: rename to *customer (creation request)*

File: apps/data-types.md [hunk 10]
```diff
@@ -83,15 +138,19 @@ This object is used for the attributes of shippingAddress and billingAddress.
 | internalNote | string | Internal note for the customer done by the merchant. |
 | links | array of [link](page:apps-data-types#link) | The link to the customer and their orders. |
 
-## customAttribute
+## customerPatchRequest
```
Comment (ubep) [inline apps/data-types.md]: rename to *JsonPatch*.
this object can be used then for all patch requests

File: apps/data-types.md [hunk 11]
```diff
@@ -255,6 +314,26 @@ This object is used for the attributes of product.
...
+
+| Attribute | Type | Description |
+| - | :-: |  - |
+| productNumber | string | The product number. |
+| name | string | The name of the product. |
+| shortDescription | string | The short description of the product. |
+| description | string | The description of the product. |
+| manufacturer | string | The manufacturer of the product. |
+| price | string | The price of the product. |
+| searchKeywords | string | The search terms for the product. |
```
Comment (ubep) [inline apps/data-types.md]: missing attributes: 
- essentialFeatures
- energyLabelsString
- upc
- ean

File: apps/data-types.md (hunk 3 above)
Comment (BirgitBader) [inline apps/data-types.md]: @ubep for amount the type is *number*, isn't it inconsistent to use decimal here? Or is there definitely a difference?

File: apps/data-types.md [hunk 12]
```diff
@@ -83,15 +111,11 @@ This object is used for the attributes of shippingAddress and billingAddress.
 | internalNote | string | Internal note for the customer done by the merchant. |
 | links | array of [link](page:apps-data-types#link) | The link to the customer and their orders. |
 
-## customAttribute
+## customer (create request)
```
Comment (ubep) [inline apps/data-types.md]: add
- internalNote
- billingAddress
to customer (create request)

File: apps/data-types.md [hunk 13]
```diff
@@ -83,15 +111,11 @@ This object is used for the attributes of shippingAddress and billingAddress.
...
 
 | Attribute | Type | Description |
 | - | :-: |  - |
-| key | string | The identifier of the custom attribute. |
-| displayKey | string | The displayed name of the custom attribute. |
-| singleValue | boolean | Indicates if just one feature is selected for the custom attribute.  |
-| type | enum | The data type of the custom attribute. Can be *string*, *number*, *bool*, *datetime*, *time*, *url*.|
-| values | array of [variationValue](page:apps-data-types#variationvalue)| The options selected for the custom attribute. |
+| customerNumber | string | The number by which the customer is tracked.|
```
Comment (ubep) [inline apps/data-types.md]: mention that customerNumber is optional. If not provided, EP6 will generate one.

File: apps/data-types.md [hunk 14]
```diff
@@ -109,6 +133,14 @@ This object is used for the attributes of images.
 | name | string | The name of the slideshow image. |
 | sizes | array of [image](page:apps-data-types#image) | The size of the images in the slideshow. |
 
+## jsonPatch
+
+| Attribute | Type | Description |
+| - | :-: |  - |
+| op | string | Patch operation to perform. Allows `add` and `remove`. *customerNumber* supports `add` only.|
```
Comment (ubep) [inline apps/data-types.md]: remove customer-specific description from generic jsonPatch request object

**Comment by ubep** [comment]
request objects for legal pages missing

File: apps/data-types.md [hunk 15]
```diff
@@ -240,6 +264,36 @@ This object is used for the attributes of product.
 | stocklevel | string | Only available with the `products_write` authorisation. Indicates the stocklevel of the product. |
 | links | array of [link](page:apps-data-types#link) | The links to the product and product category. |
 
+## product (create request)
```
Comment (ubep) [inline apps/data-types.md]: maybe rename to
*product (create or update request)*

**Pull request merged**